### DISCUSSION TITLE
is it possible to change how default table name are generated from the class name when using ORM ?

### DISCUSSION THREAD

**Discussion by 12rambau** [fastapi/sqlmodel#1031] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
toto
```

### Description

I'm starting with the ORM and I try y best to right as few things as possible and hand over complex things to the lib. 

I would like to create the following class: 

```python 
class CropMask(Base, table=True):
    """Cropmask used to filter data when loading indicators from GEE.

    All the cropmasks are exported as images in GEE to accelerate computation.
    """
    id: Optional[int] = Field(default=None, primary_key=True)

    name: str
    "technical name of the asset. Use snake case like 'esa_crop'"

    asset_id: str
    "the binary asset to use as a mask image"

    geolocations: list["Geolocation"] = Relationship(back_populates="crop_mask")
    "The geolocations that use this cropmask"
```

and add a relation with the geolocation class as : 

```python 
class Geolocation(Base, table=True):
    """The geolocation as defined in MDM.

    The Geolocation are aggregation of multiple administrative areas from GADM. They are used to reduce the amount of reducers to compute when downloading an indicator from GEE.
    For example we don't need to download all France département when computing Wheat related indicators, only the ones where Wheat is grown.
    The values of this table should be populated from MDM sources at building stage of the DB to match company description of geometries.
    """

    id: Optional[int] = Field(default=None, primary_key=True)

    gid_ldc: str
    "The name of the geolocation"

    display_name: str
    "The name use by platform member to name this aggregation"

    crop_mask_id: Optional[int] = Field(default=None, foreign_key="crop_mask.id")
    crop_mask: Optional[CropMask] = Relationship(back_populates="geolocations")
    "The cropmask used to filter the data when computing indicators"
```

In the `Field` definition, I need to use the table name and if I'm correct this name is generated from the class name by default. The default behaviour seems to be `CropMask` -> `cropmask` but I would prefer to have snake case names. AS I would this behaviour to be available to all my classe, is there a way to modify how the `SQLModel` generate table names when the tablename is missing ? 

### Operating System

Linux

### SQLModel Version

0.0.19

### Python Version

3.10

**Comment by YuriiMotov** [comment, marked as answer]
```py
from typing import Optional

from pydantic.alias_generators import to_snake
from sqlalchemy.orm import declared_attr
from sqlmodel import Field, SQLModel, create_engine

class SQLModelBase(SQLModel):
    @declared_attr.directive  # type: ignore[misc]
    @classmethod
    def __tablename__(cls) -> str:
        return to_snake(cls.__name__)

class CropMask(SQLModelBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)

engine = create_engine("sqlite:///", echo=True)
SQLModel.metadata.create_all(engine)

```


Output:
```
CREATE TABLE crop_mask (
        id INTEGER NOT NULL, 
        PRIMARY KEY (id)
)
```
